evalml/pipelines/components/estimators/regressors/arima_regressor.py

```diff
@@ -129,6 +132,23 @@ def _set_forecast(self, X):
         fh_ = ForecastingHorizon([i + 1 for i in range(len(X))], is_relative=True)
         return fh_
 
+    def _get_sp(self, X):
+        if X is None:
+            return 1
+        freq_mappings = {
+            "D": 7,
+            "W": 52,
+            "M": 12,
+            "Q": 4,
+        }
+        time_index = self._parameters.get("time_index", None)
+        sp = self.arima_parameters["sp"]
+        if sp == "detect":
+            inferred_freqs = X.ww.infer_temporal_frequencies()
```

Comment:
So, dumb question here: if X already has its freqs inferred, will this reinfer them?  Inference of data types is expensive, I'd assume freq inference is, too.

Reply:
Unfortunately, from what I can tell, neither woodwork nor pandas saves the frequency information - you can run inference in either library to get the frequency, but it's not saved anywhere between runs.